examples/jsm/interactive/HTMLMesh.js

```diff
@@ -268,25 +283,120 @@ function html2canvas( element ) {
 
 			const backgroundColor = style.backgroundColor;
 
+			// Get the border of the element used for fill and border
+			roundRectPath(x, y, width, height, parseFloat(style.borderRadius) );
 			if ( backgroundColor !== 'transparent' && backgroundColor !== 'rgba(0, 0, 0, 0)' ) {
 
 				context.fillStyle = backgroundColor;
-				context.fillRect( x, y, width, height );
-
+				context.fill();
 			}
 
-			drawBorder( style, 'borderTop', x, y, width, 0 );
-			drawBorder( style, 'borderLeft', x, y, 0, height );
-			drawBorder( style, 'borderBottom', x, y + height, width, 0 );
-			drawBorder( style, 'borderRight', x + width, y, 0, height );
-
-			if ( element.type === 'color' || element.type === 'text' || element.type === 'number' ) {
-
-				clipper.add( { x: x, y: y, width: width, height: height } );
+			// If all the borders match then stroke the round rectangle
+			const borders = ['borderTop', 'borderLeft', 'borderBottom', 'borderRight'];
+			let match = true;
+			let prevBorder = null;
+			for (const border of borders) {
+				if (prevBorder) {
+					match = match && style[ border + 'Width' ] && style[ border + 'Color' ] && style[ border + 'Style'];
```

Comment:
It seems `match &&` can be removed since it always evaluates to `true`.

https://deepscan.io/dashboard/#view=project&pid=19901&iid=119969579&tid=16600&subview=pull-request&prid=1224310

Reply:
I don't think that is true because if any of the conditions don't match it should all become false.

Reply:
Oh this is a mistake I don't perform the comparison I'll fix it.

Reply:
Fixed

Reply:
`match &&` is still unneeded because the loop immediately breaks if `match` is false.

Reply:
Yeah your right.